go/pipeline/base_batch.go

```diff
@@ -0,0 +1,6454 @@
+//
+// Command Response:
+//
+//	The number of elements in the resulting sorted set at destination.
+//
+// [valkey.io]: https://valkey.io/commands/zunionstore/
+func (b *BaseBatch[T]) ZUnionStoreWithOptions(
+	destination string,
+	keysOrWeightedKeys options.KeysOrWeightedKeys,
+	zUnionOptions *options.ZUnionOptions,
+) *T {
+	args, err := keysOrWeightedKeys.ToArgs()
+	if err != nil {
+		return b.addError("ZUnionStoreWithOptions", err)
+	}
+	args = append([]string{destination}, args...)
+	if zUnionOptions != nil {
+		optionsArgs, err := zUnionOptions.ToArgs()
+		if err != nil {
+			return b.addError("ZUnionStoreWithOptions", err)
+		}
+		args = append(args, optionsArgs...)
+	}
+	return b.addCmdAndTypeChecker(C.ZUnionStore, args, reflect.Int64, false)
+}
+
+// Pops one or more member-score pairs from the first non-empty sorted set,
+// with the given keys being checked in the order provided.
+//
+// See [valkey.io] for details.
+//
+// Parameters:
+//
+//	keys - An array of keys to sorted sets.
+//	scoreFilter - The element pop criteria - either [options.MIN] or [options.MAX] to pop members with the lowest/highest
+//
+// scores.
+//
+// Command Response:
+//
+//	A `models.KeyWithArrayOfMembersAndScores` struct containing:
+//	- The key from which the elements were popped.
+//	- An array of member-score pairs of the popped elements.
+//	  Returns `nil` if no member could be popped.
+//
+// [valkey.io]: https://valkey.io/commands/zmpop/
+func (b *BaseBatch[T]) ZMPop(keys []string, scoreFilter constants.ScoreFilter) *T {
+	scoreFilterStr, err := scoreFilter.ToString()
+	if err != nil {
+		return b.addError("ZMPop", err)
+	}
+
+	// Check for potential length overflow.
+	if len(keys) > math.MaxInt-2 {
+		return b.addError("ZMPop", &errors.RequestError{
+			Msg: "Length overflow for the provided keys",
+		})
+	}
+
+	// args slice will have 2 more arguments with the keys provided.
+	args := make([]string, 0, len(keys)+2)
+	args = append(args, strconv.Itoa(len(keys)))
+	args = append(args, keys...)
+	args = append(args, scoreFilterStr)
+	return b.addCmdAndTypeChecker(C.ZMPop, args, reflect.Map, true)
+}
+
+// Removes and returns up to `count` members from the first non-empty sorted set
+// among the provided `keys`, based on the specified `scoreFilter` criteria.
+//
+// See [valkey.io] for details.
+//
+// Parameters:
+//
+//	keys - An array of keys to sorted sets.
+//	scoreFilter - The element pop criteria - either [options.MIN] or [options.MAX] to pop members with the lowest/highest
+//
+// scores.
+//
+//	options - Pop options, see [options.ZMPopOptions].
+//
+// Command Response:
+//
+//	An object containing the key name of the set from which the element was popped and
+//	an array of member scores of the popped elements.
+//	Returns `nil` if no member could be popped.
+//
+// [valkey.io]: https://valkey.io/commands/zmpop/
+func (b *BaseBatch[T]) ZMPopWithOptions(keys []string, scoreFilter constants.ScoreFilter, options options.ZMPopOptions) *T {
```

Comment:
This differs from the signature for ZMPopWithOptions in baseClient - specifically the options parameter is a different type.